File: backend/Entities/Oportunidad.cs
using System;
using System.Collections.Generic;

namespace RebelQueenBack.Core.Entities
{
    /// <summary>
    /// Entidad de Oportunidad de Licitación en Perú Compras
    /// </summary>
    public class Oportunidad
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        // ── 1. DATOS DE LA CONVOCATORIA PERÚ COMPRAS (INMUTABLES TRAS LA CREACIÓN) ──
        public string NumeroRequerimiento { get; set; } = string.Empty; // Ej: REQ-001
        
        public Guid AcuerdoMarcoId { get; set; }
        public virtual AcuerdoMarco AcuerdoMarco { get; set; } = null!;

        public DateTime? FechaVencimientoLicitacion { get; set; }

        // ── 2. MARCAS Y PRODUCTOS / LÍMITES (EDITABLES) ──
        public virtual ICollection<OportunidadMarca> OportunidadMarcas { get; set; } = new List<OportunidadMarca>();
        public virtual ICollection<OportunidadProducto> Productos { get; set; } = new List<OportunidadProducto>();

        // Límite total calculado: Sum(Cantidad * LimiteUnitario)
        public decimal LimiteTotal { get; set; }

        // Estado: En Licitación, Por Vencer, Cotizada, Adjudicada, Desestimada
        public string Estado { get; set; } = "En Licitación";

        // Auditoría
        public string? CreadoPorUsuarioId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: src/api/Dtos/Oportunidad.ts
import type { AcuerdoMarco } from './AcuerdoMarco';
import type { Marca } from './Marca';
import type { ProductoItem } from './ProductoItem';


/* ── Modelo de Oportunidad ── */
export interface Oportunidad {
    id: string | number;
    numeroRequerimiento: string;
    acuerdoMarco: AcuerdoMarco;
    empresaId?: number | null;
    empresaRazonSocial?: string;
    empresaRuc?: string;
    entidadConvocante?: string;
    marcas: Marca[];
    fechaVencimiento: string;
    items: ProductoItem[];
    limiteTotal: number;
    estado: 'En Licitación' | 'Cotizada' | 'Adjudicada' | 'Desestimada';
    creadoPor: string;
    createdAt: string;
    updatedAt?: string;
}